connectors/src/connectors/confluence/lib/confluence_client.ts

```diff
@@ -218,6 +224,24 @@ export class ConfluenceClient {
       if (response.status === 403 && response.statusText === "Forbidden") {
         throw new ExternalOAuthTokenError();
       }
+      // retry the request after a delay: https://developer.atlassian.com/cloud/confluence/rate-limiting/
+      if (response.status === 429) {
+        if (retryCount < MAX_RATE_LIMIT_RETRY_COUNT) {
+          const delayMs =
+            Number(response.headers?.get("Retry-After") || 10) * 1000;
```

Comment:
WE should never rely on `Number` to parse numbers in JS because it handles some symbols like (`Infinity`).

Here you want to use a mix of `parseInt(x, 10) and `Number.isNan()`

```
> Number(Infinity)
Infinity
> parseInt(Infinity, 10)
NaN
```